### PR TITLE
[tgui] Update TGUI to 0.9

### PR DESCRIPTION
**Pull request by rubenwardy**
**Describe the pull request**

- What does your PR fix? TGUI is outdated - 0.9 is now available
- Which triplets are supported/not supported? Have you updated the CI baseline? Same as before
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

**Review by NancyLi1013 — COMMENTED** [review]
Could you please use `vcpkg_check_features()` to handle the feature?

Please refer to 
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_check_features.mdhttps://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_check_features.md.

**Comment by rubenwardy** [comment]
Updated

File: ports/tgui/CONTROL [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-Version: 0.8.8
+Version: 0.9.1
 Description: TGUI is an easy to use, cross-platform, C++ GUI for SFML.
 Build-Depends: sfml
 
 Feature: tool
 Description: Build GUI builder
+
+Feature: shared
+Description: Build shared libs
```
Comment (NancyLi1013) [inline ports/tgui/CONTROL:10]:
```suggestion
```
This should not be a feature. It should be built always.

File: ports/tgui/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
-    SHA512 52d044e90fde0eb6c46c3af3e11a6c8641cce301e863858701aaa909b945c76a1dade2f96a2176a223d92c51f84fe8a6dd883ec74873e8d2fcf09dac50869891
-    HEAD_REF 0.8
+    REF 588362edc42772458fc232c5375a370aa7c449e1 # v0.9.1
+    SHA512 3101be4c3f70c8cf6ea7a880fb97608c5fabb33ca14d87882efaf0a270eac3e594dc4afd9011962b290361ad2579d11e11d5157eda5163c3a3d00e41e8774f23
+    HEAD_REF 0.9
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        shared TGUI_SHARED_LIBS
```
Comment (NancyLi1013) [inline ports/tgui/portfile.cmake:11]:
```suggestion
```

File: ports/tgui/portfile.cmake [hunk 3]
```diff
@@ -13,27 +19,20 @@ set(TGUI_TOOLS_PATH ${CURRENT_PACKAGES_DIR}/tools/tgui)
...
-
 vcpkg_configure_cmake(
     SOURCE_PATH "${SOURCE_PATH}"
     DISABLE_PARALLEL_CONFIGURE
     PREFER_NINJA
     OPTIONS
-        -DTGUI_BUILD_GUI_BUILDER=${BUILD_GUI_BUILDER}
+        ${FEATURE_OPTIONS}
         -DTGUI_MISC_INSTALL_PREFIX=${TGUI_SHARE_PATH}
-        -DTGUI_SHARED_LIBS=${TGUI_SHARED_LIBS}
```
Comment (NancyLi1013) [inline ports/tgui/portfile.cmake:29]: This way is more proper to build dynamic library. So it is still needed here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/tgui/portfile.cmake (hunk 3 above)
Comment (rubenwardy) [reply ports/tgui/portfile.cmake:29]: I don't understand what this does? Where does it get that variable from?

File: ports/tgui/CONTROL (hunk 1 above)
Comment (rubenwardy) [reply ports/tgui/CONTROL:10]: According to the author, to build both shared and static you need to build TGUI twice

Also worth noting that I'm unable to get the shared version to work, it seems to link incorrectly with OpenGL. I'll revert this

File: ports/tgui/portfile.cmake (hunk 3 above)
Comment (NancyLi1013) [reply ports/tgui/portfile.cmake:29]: It is from `string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" TGUI_SHARED_LIBS)`.
When building dynamic build, `TGUI_SHARED_LIBS` will be set as `ON`.
Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Have you tested the feature `tool`? Can `tgui[tool]` build fine?

**Comment by rubenwardy** [comment]
It fails with errors linking to OpenGL: https://gist.github.com/rubenwardy/682f8da96fec8cf2bbb8b9860977af27

I suspect that this is a problem with the SFML port, because I get the same error when building with just SFML from vcpkg. Here's the minimal reproducable example, a work around is to link explicitly with OpenGL. https://gist.github.com/rubenwardy/66fa8a9831476954db1e7a873ca31773#file-cmakelists-txt-L8

Perhaps something is borked with how SFML is declaring dependencies?

**Comment by NancyLi1013** [comment]
Hi @rubenwardy 

Thanks for your reply and investigation. I noticed that you tested the triplet x64-linux.

I have checked that it also failed with the same error on current master branch. So this is a known problem, which doesn't block the update.

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your update @rubenwardy.

We might fix the feature in another PR.

File: ports/tgui/portfile.cmake [hunk 4]
```diff
@@ -54,7 +53,9 @@ if(BUILD_GUI_BUILDER)
 endif()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share)
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/TGUI/nanosvg")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/TGUI/extlibs/nanosvg")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/TGUI/extlibs/stb")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/TGUI/extlibs/glad")
```
Comment (vicroms) [inline ports/tgui/portfile.cmake:58]: I'm concerned about this lines. Is `tgui` building these external libraries or depends on them on any way? 
If so, all of these are already available as vcpkg ports so `tgui` should link these libraries from vcpkg instead.

**Comment by NancyLi1013** [comment]
Related #17093

File: ports/tgui/portfile.cmake (hunk 4 above)
Comment (texus) [reply ports/tgui/portfile.cmake:58]: These are all header-only libraries that are mostly needed when building the library. Several of the extlibs folders can be removed without breaking user code.
I'm not sure whether linking to the vcpkg libraries is going to work, some of these extlibs are modified versions.

There are currently 4 extlibs:
- Aurora is exposed in public headers and is the only extlib that can't be removed
- glad is included in a public header file, but this header file is undocumented so it is unlikely to be included in user code
- nanosvg is only included internally
- stb is only included internally

**Comment by rubenwardy** [comment]
I don't have time for this, and the other PR has been merged so maybe it has been fixed

**Pull request closed without merging**